api/base/serializers.py

```diff
@@ -225,7 +225,7 @@ def to_internal_value(self, data):
             if request.method in utils.UPDATE_METHODS and not utils.is_bulk_request(request):
                 id_field = getattr(self.root.instance, self.source, '_id')
                 if id_field != data:
-                    raise Conflict()
+                    raise Conflict(detail=('The resource id you specified "' + data + '" does not match the id of the resource you specified "' + id_field + '".'))
```

Comment:
I think this could be more useful. In this case, one of these comes from the URL and the other comes from the JSON, yes? So maybe, `The id you used in the URL "axkrt" does not match the id you used in the json body's id field "axktr".`